packages/react-reconciler/src/ReactFiberReconcilerTypes.js

```diff
@@ -0,0 +1,369 @@
+  // This is also used to quickly determine if a subtree has no pending changes.
+  expirationTime: ExpirationTime, // This is a pooled version of a Fiber. Every fiber that gets updated will
+  // eventually have a pair. There are cases when we can clean up pairs to save
+  // memory if we need to.
+  alternate: Fiber | null, // Conceptual aliases
+  // workInProgress : Fiber ->  alternate The alternate used for reuse happens
+  // to be the same as work in progress.
+  // __DEV__ only
+  _debugID?: number,
+  _debugSource?: Source | null,
+  _debugOwner?: Fiber | null,
+  _debugIsCurrentlyTiming?: boolean,
+|};
+type PartialState<State, Props> =
+  | $Subtype<State>
+  | ((prevState: State, props: Props) => $Subtype<State>); // Callbacks are not validated until invocation
+type Callback = mixed;
+type Update<State> = {
+  expirationTime: ExpirationTime,
+  partialState: PartialState<any, any>,
+  callback: Callback | null,
+  isReplace: boolean,
+  isForced: boolean,
+  next: Update<State> | null,
+};
+type UpdateQueue<State> = {
+  // A processed update is not removed from the queue if there are any
+  // unprocessed updates that came before it. In that case, we need to keep
+  // track of the base state, which represents the base state of the first
+  // unprocessed update, which is the same as the first update in the list.
+  baseState: State, // For the same reason, we keep track of the remaining expiration time.
+  expirationTime: ExpirationTime,
+  first: Update<State> | null,
+  last: Update<State> | null,
+  callbackList: Array<Update<State>> | null,
+  hasForceUpdate: boolean,
+  isInitialized: boolean, // Dev only
+  isProcessing?: boolean,
+};
+type Batch = {
+  _defer: boolean,
+  _expirationTime: ExpirationTime,
+  _onComplete: () => mixed,
+  _next: Batch | null,
+};
+type FiberRoot = {
+  containerInfo: any, // Used only by persistent updates.
+  pendingChildren: any, // The currently active root fiber. This is the mutable root of the tree.
+  current: Fiber,
+  remainingExpirationTime: ExpirationTime, // Determines if this root can be committed.
+  isReadyForCommit: boolean, // A finished work-in-progress HostRoot that's ready to be committed.
+  // TODO: The reason this is separate from isReadyForCommit is because the
+  // FiberRoot concept will likely be lifted out of the reconciler and into
+  // the renderer.
+  finishedWork: Fiber | null, // Top context object, used by renderSubtreeIntoContainer
+  context: Object | null,
+  pendingContext: Object | null, // Determines if we should attempt to hydrate on the initial mount
+  +hydrate: boolean, // List of top-level batches. This list indicates whether a commit should be
+  // deferred. Also contains completion callbacks.
+  // TODO: Lift this into the renderer
+  firstBatch: Batch | null, // Linked-list of roots
+  nextScheduledRoot: FiberRoot | null,
+};
+export type Deadline = {timeRemaining: () => number};
+type OpaqueHandle = Fiber;
+type OpaqueRoot = FiberRoot;
```

Comment:
these should be exported as well. They also should probably not be the actual `Fiber` type. The goal of this is that they are opaque to a user, meaning they track the right type, but the contents of the type aren't visible to a user of it, e.g react-dom-lite will pass around `OpaqueHandle` but should know or touch anything on it.